static/app/components/stream/groupCheckBox.tsx

```diff
@@ -14,28 +13,30 @@ type State = {
   isSelected: boolean;
 };
 
-const GroupCheckBox = createReactClass<Props, State>({
-  displayName: 'GroupCheckBox',
-
-  mixins: [Reflux.listenTo(SelectedGroupStore, 'onSelectedGroupChange') as any],
-
-  getInitialState() {
-    return {
-      isSelected: SelectedGroupStore.isSelected(this.props.id),
-    };
-  },
+class GroupCheckBox extends Component<Props, State> {
+  state = {
+    isSelected: SelectedGroupStore.isSelected(this.props.id),
+  };
 
   componentWillReceiveProps(nextProps: Props) {
     if (nextProps.id !== this.props.id) {
       this.setState({
         isSelected: SelectedGroupStore.isSelected(nextProps.id),
       });
     }
-  },
+  }
 
   shouldComponentUpdate(_nextProps, nextState) {
     return nextState.isSelected !== this.state.isSelected;
-  },
+  }
+
+  componentWillUnmount() {
+    this.listener();
+  }
+
+  listener = SelectedGroupStore.listen(() => {
```

Comment:
can we rename this `unsubscribe` or something? The result of calling listen is an unsub function, no? I did a double take on the name here when reading the code.